src/ImageSharp/Image.cs

```diff
@@ -140,15 +148,20 @@ public Image<TPixel2> CloneAs<TPixel2>()
         protected void UpdateSize(Size size) => this.size = size;
 
         /// <summary>
-        /// Disposes the object and frees resources for the Garbage Collector.
+        /// Internal routine for freeing managed resources called from <see cref="Dispose"/>
         /// </summary>
-        /// <param name="disposing">Whether to dispose of managed and unmanaged objects.</param>
-        protected abstract void Dispose(bool disposing);
+        protected abstract void DisposeManaged();
 
         /// <summary>
         /// Throws <see cref="ObjectDisposedException"/> if the image is disposed.
         /// </summary>
-        internal abstract void EnsureNotDisposed();
+        internal void EnsureNotDisposed()
+        {
+            if (this.isDisposed)
+            {
+                throw new ObjectDisposedException("Trying to execute an operation on a disposed image.");
```

Comment:
We should be using a ThrowHelper here to allow inlining on sealed instances.

Reply:
You mean ThrowHelper class from ThrowHelper.cs? Should I add ThrowArgumentException method or am I missing something? (there is none as far as I can see :P)